Core/include/Acts/Utilities/HelpersMatrixGenerated.ipp

```diff
@@ -0,0 +1,825 @@
+  pC[61] = pA[5] * pB[56] + pA[13] * pB[57] + pA[21] * pB[58] +
+           pA[29] * pB[59] + pA[37] * pB[60] + pA[45] * pB[61] +
+           pA[53] * pB[62] + pA[61] * pB[63];
+  pC[6] = pA[6] * pB[0] + pA[14] * pB[1] + pA[22] * pB[2] + pA[30] * pB[3] +
+          pA[38] * pB[4] + pA[46] * pB[5] + pA[54] * pB[6] + pA[62] * pB[7];
+  pC[14] = pA[6] * pB[8] + pA[14] * pB[9] + pA[22] * pB[10] + pA[30] * pB[11] +
+           pA[38] * pB[12] + pA[46] * pB[13] + pA[54] * pB[14] +
+           pA[62] * pB[15];
+  pC[22] = pA[6] * pB[16] + pA[14] * pB[17] + pA[22] * pB[18] +
+           pA[30] * pB[19] + pA[38] * pB[20] + pA[46] * pB[21] +
+           pA[54] * pB[22] + pA[62] * pB[23];
+  pC[30] = pA[6] * pB[24] + pA[14] * pB[25] + pA[22] * pB[26] +
+           pA[30] * pB[27] + pA[38] * pB[28] + pA[46] * pB[29] +
+           pA[54] * pB[30] + pA[62] * pB[31];
+  pC[38] = pA[6] * pB[32] + pA[14] * pB[33] + pA[22] * pB[34] +
+           pA[30] * pB[35] + pA[38] * pB[36] + pA[46] * pB[37] +
+           pA[54] * pB[38] + pA[62] * pB[39];
+  pC[46] = pA[6] * pB[40] + pA[14] * pB[41] + pA[22] * pB[42] +
+           pA[30] * pB[43] + pA[38] * pB[44] + pA[46] * pB[45] +
+           pA[54] * pB[46] + pA[62] * pB[47];
+  pC[54] = pA[6] * pB[48] + pA[14] * pB[49] + pA[22] * pB[50] +
+           pA[30] * pB[51] + pA[38] * pB[52] + pA[46] * pB[53] +
+           pA[54] * pB[54] + pA[62] * pB[55];
+  pC[62] = pA[6] * pB[56] + pA[14] * pB[57] + pA[22] * pB[58] +
+           pA[30] * pB[59] + pA[38] * pB[60] + pA[46] * pB[61] +
+           pA[54] * pB[62] + pA[62] * pB[63];
+  pC[7] = pA[7] * pB[0] + pA[15] * pB[1] + pA[23] * pB[2] + pA[31] * pB[3] +
+          pA[39] * pB[4] + pA[47] * pB[5] + pA[55] * pB[6] + pA[63] * pB[7];
+  pC[15] = pA[7] * pB[8] + pA[15] * pB[9] + pA[23] * pB[10] + pA[31] * pB[11] +
+           pA[39] * pB[12] + pA[47] * pB[13] + pA[55] * pB[14] +
+           pA[63] * pB[15];
+  pC[23] = pA[7] * pB[16] + pA[15] * pB[17] + pA[23] * pB[18] +
+           pA[31] * pB[19] + pA[39] * pB[20] + pA[47] * pB[21] +
+           pA[55] * pB[22] + pA[63] * pB[23];
+  pC[31] = pA[7] * pB[24] + pA[15] * pB[25] + pA[23] * pB[26] +
+           pA[31] * pB[27] + pA[39] * pB[28] + pA[47] * pB[29] +
+           pA[55] * pB[30] + pA[63] * pB[31];
+  pC[39] = pA[7] * pB[32] + pA[15] * pB[33] + pA[23] * pB[34] +
+           pA[31] * pB[35] + pA[39] * pB[36] + pA[47] * pB[37] +
+           pA[55] * pB[38] + pA[63] * pB[39];
+  pC[47] = pA[7] * pB[40] + pA[15] * pB[41] + pA[23] * pB[42] +
+           pA[31] * pB[43] + pA[39] * pB[44] + pA[47] * pB[45] +
+           pA[55] * pB[46] + pA[63] * pB[47];
+  pC[55] = pA[7] * pB[48] + pA[15] * pB[49] + pA[23] * pB[50] +
+           pA[31] * pB[51] + pA[39] * pB[52] + pA[47] * pB[53] +
+           pA[55] * pB[54] + pA[63] * pB[55];
+  pC[63] = pA[7] * pB[56] + pA[15] * pB[57] + pA[23] * pB[58] +
+           pA[31] * pB[59] + pA[39] * pB[60] + pA[47] * pB[61] +
+           pA[55] * pB[62] + pA[63] * pB[63];
```

Comment:
You may want to cross-check the effect of this on compilation overhead (in time and memory) as compilers have a hard time churning through a lot of code. Some compilation algorithms have highly nonlinear overhead wrt input size.

Also, C compilers by default force summation to occur in the order written in the code, which is very unlikely to be optimal here. This is, in particular, very likely to hamper good code autovectorization. So you may want to disable this mechanism by compiling affected compilation units with `-ffast-math`, just to see how much the performance would change with a more optimal (according to the compiler) summation order.